src/main.rs

```diff
@@ -107,6 +106,15 @@ fn app() -> Router<SpfChecker> {
     Router::new()
         .route("/health", get(health))
         .route("/api/v1/check-spf", get(check_spf))
+        .route("/ui", get(serve_ui))
+}
+
+async fn serve_ui() -> impl IntoResponse {
```

Comment:
How about we move the HTML file into it's own folder like `src/html/`? Your code in general works, but the axum example code is using the `Html` response instead, which shortens the handler function:

```rust
async fn serve_ui() -> Html<&'static str> {
    Html(include_str!("html/ui.html"))
}
```